### ISSUE TITLE
Remove obsolete Issuer and Principal classes?

### ISSUE THREAD

**Issue by richardfan1126** [sigstore/fulcio#1798]
**Question**

* Given the introduction of new CiProvider in https://github.com/sigstore/fulcio/pull/1729
   And the fact that all CI providers have been changed to `type: ci-provider` in https://github.com/sigstore/fulcio/blob/main/config/identity/config.yaml

---

Should we remove the obsolete Issuer and Principal classes under `pkg/identity` to avoid confusion?

**Comment by haydentherapper** [comment]
Yes, we should remove these, but I would prefer to hold off until we cut a major release of Fulcio to fully remove these as I don't want to break any existing deployments.

In the meantime, we could mark these providers as [deprecated](https://go.dev/wiki/Deprecated).
